### ISSUE TITLE
Option to add a random number to each requests' query string

### ISSUE THREAD

**Issue by underyx** [rakyll/hey#35]
I'm trying to load test my live API, but I have caching in front of it. If `hey` could add a random number to each query, that would bypass the cache, giving me realistic results.

**Comment by vthiery** [comment]
A suggestion, inspired by [cameronpoe/tsunami](https://github.com/cameronpoe/tsunami), would be to allow randomising some parts of the url, or the body. E.g. 
`./hey -c 10 -n 1000 -m GET http://localhost:8000/{[a-f0-9]-[a-z]}`
same for a POST body.

**Comment by joyrexus** [comment]
[blast's payload templates](https://github.com/dave/blast#templates) are really handy for randomizing request paths or query params.  Strikes me as a critical feature for genuine load testing of production systems given various forms of caching that can come into play.

**Comment by preslavrachev** [comment]
I have provided my vision of how this could work in the form of a PR. Please, have look and merge if happy.

**Comment by preslavrachev** [comment]
Anyone still interested in having this feature? @joyrexus @vthiery @underyx cc @rakyll

**Comment by vthiery** [comment]
@preslavrachev sure, it'd still be a nice to have feature

**Comment by gdmello** [comment]
I came here looking for a similar feature. It would be great to be able to add dynamic headers, where similar to the random number in the url, the random number could be added to an HTTP header itself.

**Comment by preslavrachev** [comment]
@gdmello Absolutely, the logic will be the same. I wanted to do it step by step though, first tackling the GET params, because I needed the functionality for my job.

What do the rest think? Should I try and add header shuffling to the same PR?

**Comment by vcidst** [comment]
Is there any update on this issue?

**Comment by dror-weiss** [comment]
I was just looking for the same feature; random json body.
This lack of support forces me to find another tool to use.
